File: src/net/eduard/api/command/SetXPCommand.java

package net.eduard.api.command;

import net.eduard.api.lib.modules.Extra;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import net.eduard.api.lib.manager.CommandManager;

public class SetXPCommand extends CommandManager {

	public SetXPCommand() {
		super("setexperience");
		 
	}
	
	
	@Override
	public boolean onCommand(CommandSender sender, Command command,
			String label, String[] args) {
		if (sender instanceof Player) {
			Player p = (Player) sender;
			if (args.length == 0) {
				return false;
			} else {
				Integer amount = Extra.toInt(args[0]);
				p.setTotalExperience(0);
				p.setExp(0);
				p.setLevel(0);
				p.giveExp(amount);
				p.sendMessage("§aSua xp foi alterada para: §2"+amount);
				p.sendMessage("§aSeu novo nível é: §2"+p.getLevel());
				p.sendMessage("§aSua barra de XP: §2"+ p.getExp());
				
				
			}
		}
		return true;
	}

}

File: src/main/java/net/eduard/api/command/SetXPCommand.kt
package net.eduard.api.command

import net.eduard.api.lib.manager.CommandManager
import net.eduard.api.lib.modules.Extra
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class SetXPCommand : CommandManager("setexperience","setexp","setxp") {

    init{
        description= "Defina a experiencia (EXP) do jogador"
        usage= "/<command> <xp>"
    }

    override fun playerCommand(player: Player, args: Array<String>) {
        if (args.isEmpty()) {
            sendUsage(player)
            return
        }
        var amount = Extra.fromMoneyToDouble(args[0])
        player.totalExperience = 0
        player.exp = 0f
        player.level = 0
        if (amount > Int.MAX_VALUE) {
            amount = Int.MAX_VALUE.toDouble()
        }
        player.giveExp(amount.toInt())
        player.sendMessage("§bSua xp foi alterada para: §3$amount")
        player.sendMessage("§bSeu novo nível é: §3" + player.level)
        player.sendMessage("§bSua barra de XP: §3" + player.exp)
    }

}